lib/internal/Magento/Framework/Model/ResourceModel/Db/Collection/AbstractCollection.php

```diff
@@ -350,7 +350,8 @@ public function addExpressionFieldToSelect($alias, $expression, $fields)
 
         $fullExpression = $expression;
         foreach ($fields as $fieldKey => $fieldItem) {
-            $fullExpression = str_replace('{{' . $fieldKey . '}}', $fieldItem, $fullExpression);
+            $quotedFieldItem =  $this->getConnection()->quote($fieldItem);
```

Comment:
Probably there should be something like this:
```suggestion
            $quotedFieldItem =  $this->getConnection()->quoteIdentifier($fieldItem);
```